apps/labs/posts/automated-accessibility-testing-with-bokeh.md

```diff
@@ -0,0 +1,212 @@
+had first-class support for the testing tools and ported to that. It was just
+before switching to Javascript equivalents that I made an important discovery:
+Due to the very nature of Bokeh, axe-core would be useless for lots of plots.
+This is because the visualizations end up getting wrapped in an HTML canvas with
+some styling in the [Shadow
+DOM](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_shadow_DOM),
+which is something axe does not know how to work with. This is a half-truth. The
+complete truth is that certain plots (or documents) rely on extra components or
+widgets like sliders and dropdowns which are rendered outside the canvas and are
+thus testable with the Axe API. This put me in a situation where I would have to
+report [my
+findings](https://github.com/bokeh/bokeh/discussions/14057#discussioncomment-10674506)
+and ask the larger Bokeh community whether they felt leaving axe-core was
+worthwhile.
+
+All of this was happening while I was simultaneously facing my biggest stumbling
+block yet: "the python in the room".
+
+## The Python in the room
+
+This was no elephant, and it certainly wasn't python-related, but it was a big
+problem for me. Owing to the custom nature of all things Bokeh, I had a problem
+with getting the environment to allow axe-core run. Axe-core required core
+modules from Node, and Bokeh had a stripped away version of node for the most
+part. I tend to like solving problems by myself with minimal hand-holding, but
+this was one of those scenarios that I just couldn't seem to fix. The ways
+around the problem looked even more problematic, and I found myself questioning
+my abilities.
+
+This was one of those moments where experience makes all the difference, and my
+mentor helped me with some other parts of the codebase we could draw inspiration
+from. He found something in one of our pair programming sessions, we built
+something that worked, tested it, pushed, and he even stood guard at the GitHub
+PR comment section, answering questions he knew I might struggle to answer, and
+giving me tips on how to implement the suggestions from reviewers. It was at
+that point in time I could really say: "We'd done something truly amazing for
+Bokeh."
+
+## Where we currently are
+
+A testing environment needs tests, and to test what we'd done, we needed a
+variety of valid data visualizations. Luckily, Bokeh very generously had us over
+100 example plots in the documentation that we could test against. Testing
+against the documentation came with a few challenges: one of which was the fact
+that the documentation would always need to be built and served locally before
+we could run axe-core tests against them. Bokeh also had certain commands we
+would have to model new ones after.
+
+Overall, our efforts towards going from 0 to 1 ended up looking like this [PR on
+creating a playwright
+environment.](https://github.com/bokeh/bokeh/pull/14032/files)
+
+## Challenges
```

Comment:
Maybe it makes sense to merge this section with "the Python in the room" above and rename the section to "Challenges" and modify it as needed to make it flow and to make it cohesive.